cmd/kaspawallet/sweep.go

```diff
@@ -0,0 +1,267 @@
+		LockTime:     0,
+		Outputs:      make([]*externalapi.DomainTransactionOutput, 1), // we should always have 1 output to the toAdress
+		SubnetworkID: subnetworks.SubnetworkIDNative,
+		Gas:          0,
+		Payload:      nil,
+	}
+}
+
+func createSplitTransactionsWithSchnorrPrivteKey(
+	params *dagconfig.Params,
+	selectedUTXOs []*libkaspawallet.UTXO,
+	toAddress util.Address,
+	feePerInput int) ([]*externalapi.DomainTransaction, error) {
+
+	var splitTransactions []*externalapi.DomainTransaction
+
+	// I need to add extra mass to transaction, txmasscalculater seems to undercalculate, by around this amount.
+	extraMass := uint64(7000)
+	massCalculater := txmass.NewCalculator(params.MassPerTxByte, params.MassPerScriptPubKeyByte, params.MassPerSigOp)
+
+	//for refernece this keeps track in respect to dummyWindow[0], not dummyWindow[1]
+	currentIdxOfSplit := 0
+
+	totalAmount := uint64(0)
+
+	totalSplitAmount := uint64(0)
+
+	ScriptPublicKey, err := txscript.PayToAddrScript(toAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	dummyTransactionWindow := make([]*externalapi.DomainTransaction, 2)
+	//[0] is the last build that didn't violate mass, that can be added as a split
+	dummyTransactionWindow[0] = newDummyTransaction()
+	//[1] represents the tested unsigned transaction
+	dummyTransactionWindow[1] = newDummyTransaction()
+
+	//loop through utxos commit segments that don't violate max mass
+	for i, currentUTXO := range selectedUTXOs {
+
+		if currentIdxOfSplit == 0 {
+			dummyTransactionWindow[0] = newDummyTransaction()
+			dummyTransactionWindow[1] = newDummyTransaction()
+			//we assume a transaction without inputs, and 1 undefined output cannot violate transaction mass
+		}
+
+		currentAmount := currentUTXO.UTXOEntry.Amount()
+		totalSplitAmount = totalSplitAmount + currentAmount
+		totalAmount = totalAmount + currentAmount
+
+		dummyTransactionWindow[1].Inputs = append(
+			dummyTransactionWindow[1].Inputs,
+			&externalapi.DomainTransactionInput{
+				PreviousOutpoint: *currentUTXO.Outpoint,
+				UTXOEntry: utxo.NewUTXOEntry(
+					currentUTXO.UTXOEntry.Amount(),
+					currentUTXO.UTXOEntry.ScriptPublicKey(),
+					false,
+					constants.UnacceptedDAAScore,
+				),
+				SigOpCount: 1,
+			},
+		)
+
+		dummyTransactionWindow[1].Outputs[0] = &externalapi.DomainTransactionOutput{
+			Value:           totalSplitAmount - uint64(len(dummyTransactionWindow[1].Inputs)*feePerInput),
+			ScriptPublicKey: ScriptPublicKey,
+		}
+
+		if massCalculater.CalculateTransactionMass(dummyTransactionWindow[1])+extraMass >= mempool.MaximumStandardTransactionMass {
+			splitTransactions = append(splitTransactions, dummyTransactionWindow[0])
+			currentIdxOfSplit = 0
+			totalSplitAmount = 0
+			totalAmount = totalAmount + currentAmount
+			dummyTransactionWindow[0] = newDummyTransaction()
+			dummyTransactionWindow[1] = newDummyTransaction()
+			if i == len(selectedUTXOs)-1 {
+				splitTransactions = append(splitTransactions, dummyTransactionWindow[1])
```

Comment:
I don't get it. This is a newly created dummy transactions. How can you append it to `splitTransactions`?